bpf/bpf_test.go

```diff
@@ -761,3 +762,75 @@ func TestIPv6NotSupported(t *testing.T) {
 		t.Fatalf("creating an IPv6 blacklist should have failed")
 	}
 }
+
+func TestVersionParse(t *testing.T) {
+	t.Log("Test version parsing")
+	ubuntuVersionStr := "Linux version 5.3.0-39-generic (buildd@lcy01-amd64-016) (gcc version 9.3.0 (Ubuntu 9.3.0-10ubuntu2)) #43-Ubuntu SMP Fri Jun 19 10:28:31 UTC 2020"
+	rhelVersionStr := "Linux version 4.18.0-193.el8.x86_64 (mockbuild@x86-vm-08.build.eng.bos.redhat.com) (gcc version 8.3.1 20191121 (Red Hat 8.3.1-5) (GCC)) #1 SMP Fri Mar 27 14:35:58 UTC 2020"
+	fedVersionStr := "Linux version 5.3.0-193.el8.x86_64 (mockbuild@x86-vm-08.build.eng.bos.redhat.com) (gcc version 8.3.1 20191121 (Fedora 8.3.1-5) (GCC)) #1 SMP Fri Mar 27 14:35:58 UTC 2020"
+	distName := versionparse.GetDistFromString(ubuntuVersionStr)
+	if distName != "ubuntu" {
+		t.Fatalf("Parsing distribution name failed")
+	}
+	parsedVer, err := versionparse.GetVersionFromString(ubuntuVersionStr)
+	if err != nil {
+		t.Fatalf("Parsing kernel version failed")
+	}
+	expVer := GetMinKernelVersionForDistro(distName)
+	if parsedVer.Compare(expVer) != 0 {
+		t.Fatalf("Parsed version not same as expected version")
+	}
+
+	distName = versionparse.GetDistFromString(rhelVersionStr)
+	if distName != "rhel" {
+		t.Fatalf("Parsing distribution name failed")
+	}
+	parsedVer, err = versionparse.GetVersionFromString(rhelVersionStr)
+	if err != nil {
+		t.Fatalf("Parsing kernel version failed")
+	}
+	expVer = GetMinKernelVersionForDistro(distName)
+	if parsedVer.Compare(expVer) != 0 {
+		t.Fatalf("Parsed version not same as expected version")
+	}
+
+	distName = versionparse.GetDistFromString(fedVersionStr)
+	if distName != "default" {
+		t.Fatalf("Parsing distribution name failed")
+	}
+	parsedVer, err = versionparse.GetVersionFromString(fedVersionStr)
+	if err != nil {
+		t.Fatalf("Parsing kernel version failed")
+	}
+	expVer = GetMinKernelVersionForDistro(distName)
+	if parsedVer.Compare(expVer) != 0 {
+		t.Fatalf("Parsed version not same as expected version")
+	}
+
+	// Tests to verify the compare function in versionparse
+	ver1 := versionparse.MustParseVersion("4.18.0-193")
+	ver2 := versionparse.MustParseVersion("4.18.0-194")
+	if ver1.Compare(ver2) != -1 {
+		t.Fatalf("Comparing kernel versions 4.18.0-193 and 4.18.0-194 failed")
```

Comment:
Nothing wrong with this but I typically use the Gomega assertions even in `go test` style tests.  They're concise and they tend to give good diags: `Expect(ver1.Compare(ver2)).To(Equal(-1))`

To use them, you have to call `RegisterTestingT(t)` at the top of the function.